### PR TITLE
Basic API documentation 

### PR DESCRIPTION
**Pull request by SatyendraBanjare**
This aims at adding documentation for luagit2's basic API functions.

@Etiene mam, @ethomson  sir, I have added documentation for some modules. I have changed the `readthedocs.io` branch for luagit2 from **master** to **Documentation** . So the documentation can be seen as it is being updated at : http://luagit2.readthedocs.io/en/latest/index.html

### CODE REVIEW

File: docs/source/index.rst [hunk 1]
```diff
@@ -6,15 +6,36 @@
 Welcome to LuaGit2's documentation!
 ===================================
 
+ `Luagit2`_ is a tool to do cool git things on lua. It depends on `Libgit2`_ and
```
Comment (Etiene) [inline docs/source/index.rst]: Specify on docs which version of Lua and which version of Libgit2

File: docs/source/index.rst [hunk 2]
```diff
@@ -6,15 +6,36 @@
...
+
+ It is built using `Lua_C_Api`_ and has its tests written using `Busted`_ . It can be easily
+ installed using luarocks and Cmake/Make.
+
+*************
+Compatibility
+*************
+
+	* Linux
+	* Mac-OsX
```
Comment (Etiene) [inline docs/source/index.rst]: replace `Mac-OsX` with `macOS`

**Comment by Etiene** [comment]
All links are pointing to the HEAD version of libgit2. It should be pointed to the version of libgit2 these bindings are compatible with. It might be head at the moment, but we can't guarantee that a future update of libgit2 won't break something, so you should specify the version number and update all the links

**Comment by SatyendraBanjare** [comment]
Alright, I will do so. Thanks mam. 

I will update the links with ones that point to `v 0.27.0`. Also I will update the docs with libgit2 version used.

File: docs/source/basic/blame.rst [hunk 3]
```diff
@@ -0,0 +1,91 @@
+Blame
+=====
```
Comment (Etiene) [inline docs/source/basic/blame.rst:2]: I noticed some modules add links to the libgit2 API docs and some, like `blame`, don't. Is there a reason for that? Or could you add the links to these other modules too?

**Comment by Etiene** [comment]
Could you add an entry, based on that table that you created, about which command bindings are not yet implemented, such as git commit amend, for example?

**Comment by Etiene** [comment]
(I clicked on close accidentally)

**Comment by SatyendraBanjare** [comment]
> Could you add an entry, based on that table that you created, about which command bindings are not yet implemented, such as git commit amend, for example?

Yes mam I will do it.

File: docs/source/basic/blame.rst (hunk 3 above)
Comment (SatyendraBanjare) [reply docs/source/basic/blame.rst:2]: I will add them soon.

Comment (Etiene) [reply docs/source/basic/blame.rst:2]: Awesome, thanks!

**Pull request merged**